packages/telemetry/src/index.ts

```diff
@@ -53,47 +63,67 @@ export class Telemetry implements TelemetryApi {
     SpanProcessor,
     exporter,
     resources = {},
+    isVerbose = false,
   }: {
-    namespace?: string
+    namespace: string
     Provider: typeof BasicTracerProvider
     detectors: DetectorSync[]
     rootContextObject?: contextObject
     version: string
     SpanProcessor: typeof SimpleSpanProcessor | typeof BatchSpanProcessor
     exporter: SpanExporter
     resources?: Attributes
+    isVerbose: boolean
   }) {
     // For troubleshooting, set the log level to DiagLogLevel.DEBUG
+    // import { diag, DiagConsoleLogger, DiagLogLevel } from '@opentelemetry/api'
     // diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.ALL)
 
+    this.isVerbose = isVerbose
+
     // Setup default resources
     const resource = Resource.default().merge(
       new Resource({
         ...resources,
-        [ SemanticResourceAttributes.SERVICE_NAME ]: 'cypress-app',
+        [ SemanticResourceAttributes.SERVICE_NAME ]: SERVICE_NAME,
         [ SemanticResourceAttributes.SERVICE_NAMESPACE ]: namespace,
         [ SemanticResourceAttributes.SERVICE_VERSION ]: version,
       }),
     )
 
     // Merge resources and create a new provider of the desired type.
-    this.provider = new Provider({ resource: resource.merge(detectResourcesSync({ detectors })) })
+    this.provider = new Provider({
+      resource: resource.merge(detectResourcesSync({ detectors })),
+    })
+
+    // Setup the exporter
+    if (SpanProcessor.name === 'BatchSpanProcessor') {
+      this.provider.addSpanProcessor(new SpanProcessor(exporter, {
+        // Double the max queue size, We were seeing telemetry bursts that would result in loosing the top span.
+        maxQueueSize: 4056,
+      }))
+    } else {
+      this.provider.addSpanProcessor(new SpanProcessor(exporter))
+    }
 
-    // Setup the console exporter
-    this.provider.addSpanProcessor(new SpanProcessor(exporter))
+    // if local visualizations enabled, create composite exporter configured
+    // to send to both local exporter and main exporter
+    const honeyCombConsoleLinkExporter = new ConsoleTraceLinkExporter({
+      serviceName: SERVICE_NAME,
+      team: 'cypress',
+      environment: (process.env.CYPRESS_CONFIG_ENV || process.env.CYPRESS_INTERNAL_ENV === 'production' ? 'cypress-app' : 'cypress-app-staging'),
```

Comment:
Other places in the repo we are ref like `env.CYPRESS_CONFIG_ENV || env.CYPRESS_INTERNAL_ENV || 'development'`. Should we here as well and/or should it be
```
(process.env.CYPRESS_CONFIG_ENV === 'production' || process.env.CYPRESS_INTERNAL_ENV === 'production') ? 'cypress-app' : 'cypress-app-staging'?
```

Reply:
It should be this:

```suggestion
      environment: ((process.env.CYPRESS_CONFIG_ENV || process.env.CYPRESS_INTERNAL_ENV) === 'production' ? 'cypress-app' : 'cypress-app-staging'),
```

Reply:
addressed in: https://github.com/cypress-io/cypress/pull/26695/commits/21e5594002bf56edea603f390336e8be7038ec2d